src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -224,6 +224,14 @@ private void InnerReleaseHandle()
             }
         }
 
+        private void TryWakeup(InnerSafeCloseSocket innerSocket)
+        {
+            if ((AsyncContext == null || !AsyncContext.GetNonBlocking()) && innerSocket != null && !_underlyingHandleNonBlocking && !innerSocket.IsClosed && !innerSocket.IsInvalid)
```

Comment:
This is a fairly complicated condition and it's not clear to me exactly what it's validating.  Can you add a comment?

Also, are both the IsClosed and IsInvalid checks necessary?

Reply:
I'll check if I can simplify this. What I want is to find out if we have valid OS blocking socket. In all other cases I want to skip the shutdown and use existing behavior. There are existing test verifying we throw if dispose is called in middle of pending async operation. I did not want to change that behavior.